devtools/maven/src/main/java/io/quarkus/maven/components/Prompter.java

```diff
@@ -1,18 +1,72 @@
 package io.quarkus.maven.components;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Consumer;
 
-import org.codehaus.plexus.component.annotations.Component;
+import org.aesh.readline.Readline;
+import org.aesh.readline.ReadlineBuilder;
+import org.aesh.readline.tty.terminal.TerminalConnection;
 
 /**
  * Prompt implementation.
  *
  * @author <a href="http://escoffier.me">Clement Escoffier</a>
  */
-@Component(role = Prompter.class, instantiationStrategy = "per-lookup")
-public class Prompter extends io.quarkus.devtools.utils.Prompter {
+public class Prompter {
+
+    private static class Prompt {
+        private final String prompt;
+        private final String defaultValue;
+        private final Consumer<String> inputConsumer;
+
+        public Prompt(String prompt, String defaultValue, Consumer<String> inputConsumer) {
+            this.prompt = prompt;
+            this.defaultValue = defaultValue;
+            this.inputConsumer = inputConsumer;
+        }
+    }
+
+    private final List<Prompt> prompts = new ArrayList<>();
+
     public Prompter() throws IOException {
-        super();
     }
 
+    public Prompter addPrompt(String prompt, Consumer<String> inputConsumer) {
+        prompts.add(new Prompt(prompt, null, inputConsumer));
+        return this;
+    }
+
+    public Prompter addPrompt(String prompt, String defaultValue, Consumer<String> inputConsumer) {
+        prompts.add(new Prompt(prompt, defaultValue, inputConsumer));
+        return this;
+    }
+
+    public void collectInput() throws IOException {
+        if (prompts.isEmpty()) {
+            return;
+        }
+        final TerminalConnection connection = new TerminalConnection();
+        try {
+            read(connection, ReadlineBuilder.builder().enableHistory(false).build(), prompts.iterator());
+            connection.openBlocking();
+        } finally {
+            connection.close();
```

Comment:
@stuartwdouglas quick question. I've switched the Maven plugin's prompter from jline to Aesh (since it's already on the classpath anyway). I was wondering whether i should restore the terminal's state like you do in the DevMojo (https://github.com/quarkusio/quarkus/blob/main/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java#L445) or is it ok to simply close the connection. I am not doing anything fancy here, simply collecting the input. Thanks!

Reply:
Then you should be ok. I put the terminal in raw mode so if the process crashes I need to restore it to the original state.